packages/asset-swapper/contracts/src/HackedERC20.sol

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.6;
+
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "./GasOverhead.sol";
+
+contract HackedERC20 {
+
+    using LibERC20TokenV06 for IERC20TokenV06;
+
+    struct ShadowedAmount {
+        bool isShadowed;
+        uint256 lastTrueAmount;
+        uint256 shadowedAmount;
+    }
+
+    struct Storage {
+        mapping(address=>ShadowedAmount) shadowedBalances;
+        mapping(address=>mapping(address=>ShadowedAmount)) shadowedAllowances;
+    }
+
+    bytes32 private constant STORAGE_SLOT = 0x64fd48372774b9637ace5c8c7a951f04ea13c793935207f2eada5382a0ec82cb;
+    address private constant GAS_OVERHEAD = 0xDeF1000000000000000000000000000000001337;
+
+    // HackedERC20 also has the overhead of being Delegated to from the replaced token
+    // USDT->DelegateHackedERC20->HackedERC20
+    uint256 constant DELEGATE_CALL_OVERHEAD = 5000;
+
+    // When enabled the HackedERC20 token will shadow and track balances
+    // when disabled (default) it will call the original implementation
+    bool public _enabled = false;
```

Comment:
uhh I think this'll occasionally explode things because it'll overwrite a token's storage :boom:
You can move this into the `Storage` to avoid the issue.